plugins/woocommerce/src/Blocks/Utils/BlockTemplateUtils.php

```diff
@@ -213,32 +213,27 @@ public static function build_template_result_from_post( $post ) {
 	 */
 	public static function build_template_result_from_file( $template_file, $template_type ) {
 		$template_file = (object) $template_file;
-
-		// If the theme has an archive-products.html template but does not have product taxonomy templates
-		// then we will load in the archive-product.html template from the theme to use for product taxonomies on the frontend.
-		$template_is_from_theme = 'theme' === $template_file->source;
-		$theme_name             = wp_get_theme()->get( 'TextDomain' );
+		$theme_name    = get_stylesheet();
```

Comment:
WordPress internally uses `get_stylesheet()` for the theme name:

https://github.com/WordPress/wordpress-develop/blob/7159243c090e429a7d2a1fd2a9509e323f67a39d/src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php#L544